server/protocols/webadmin/webadmin-data-jmap/src/test/java/org/apache/james/webadmin/data/jmap/JmapDataRoutesTest.java

```diff
@@ -0,0 +1,311 @@
+
+import java.util.Optional;
+
+import org.apache.james.core.Username;
+import org.apache.james.domainlist.api.DomainList;
+import org.apache.james.jmap.api.preview.Preview;
+import org.apache.james.jmap.draft.utils.JsoupHtmlTextExtractor;
+import org.apache.james.jmap.memory.preview.MemoryMessagePreviewStore;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageManager;
+import org.apache.james.mailbox.inmemory.InMemoryMailboxManager;
+import org.apache.james.mailbox.inmemory.manager.InMemoryIntegrationResources;
+import org.apache.james.mailbox.model.ComposedMessageId;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.task.Hostname;
+import org.apache.james.task.MemoryTaskManager;
+import org.apache.james.user.memory.MemoryUsersRepository;
+import org.apache.james.util.html.HtmlTextExtractor;
+import org.apache.james.util.mime.MessageContentExtractor;
+import org.apache.james.webadmin.WebAdminServer;
+import org.apache.james.webadmin.WebAdminUtils;
+import org.apache.james.webadmin.routes.TasksRoutes;
+import org.apache.james.webadmin.utils.ErrorResponder;
+import org.apache.james.webadmin.utils.JsonTransformer;
+import org.eclipse.jetty.http.HttpStatus;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.restassured.RestAssured;
+import io.restassured.filter.log.LogDetail;
+
+class JmapDataRoutesTest {
+    private static final DomainList NO_DOMAIN_LIST = null;
+    private static final Username BOB = Username.of("bob");
+
+    private WebAdminServer webAdminServer;
+    private MemoryTaskManager taskManager;
+    private MemoryMessagePreviewStore messagePreviewStore;
+    private InMemoryMailboxManager mailboxManager;
+    private MemoryUsersRepository usersRepository;
+
+    @BeforeEach
+    void setUp() {
+        JsonTransformer jsonTransformer = new JsonTransformer();
+        taskManager = new MemoryTaskManager(new Hostname("foo"));
+
+        messagePreviewStore = new MemoryMessagePreviewStore();
+        mailboxManager = InMemoryIntegrationResources.defaultResources().getMailboxManager();
+        usersRepository = MemoryUsersRepository.withoutVirtualHosting(NO_DOMAIN_LIST);
+        MessageContentExtractor messageContentExtractor = new MessageContentExtractor();
+        HtmlTextExtractor htmlTextExtractor = new JsoupHtmlTextExtractor();
+        webAdminServer = WebAdminUtils.createWebAdminServer(
+            new TasksRoutes(taskManager, jsonTransformer),
+            new JmapDataRoutes(taskManager, new TaskFactory(
+                new MessagePreviewCorrector(usersRepository, mailboxManager, messagePreviewStore, messageContentExtractor, htmlTextExtractor)), jsonTransformer))
+            .start();
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminServer)
+            .setBasePath("/jmap")
```

Comment:
we usually try to define endpoint that are resources and jmap doesn't look like a resource to me

Reply:
How about `/jmap/messages` ?

Reply:
I don't think such a thing exist (messages are probably a top-level concept that happens to have different representations), see my suggestion https://github.com/linagora/james-project/pull/2932/commits/8a1c238603e141d2774172bffa1fbf6016c216a8#r351781603